tests/test_tempfile.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import annotations
+
+import os
+import pathlib
+import shutil
+import tempfile
+
+import pytest
+
+from anyio import (
+    NamedTemporaryFile,
+    SpooledTemporaryFile,
+    TemporaryDirectory,
+    TemporaryFile,
+    gettempdir,
+    gettempdirb,
+    mkdtemp,
+    mkstemp,
+)
+
+pytestmark = pytest.mark.anyio
+
+
+class TestTemporaryFile:
+    async def test_temporary_file(self) -> None:
+        data = b"temporary file data"
+        async with TemporaryFile[bytes]() as af:
+            await af.write(data)
+            await af.seek(0)
+            result = await af.read()
+
+        assert result == data
+        assert af.closed
+
+
+class TestNamedTemporaryFile:
+    async def test_named_temporary_file(self) -> None:
+        data = b"named temporary file data"
+        async with NamedTemporaryFile[bytes]() as af:
+            filename = str(af.name)
```

Comment:
Redundant conversion: af.name is already a string.
```suggestion
            filename = af.name
```